### PR TITLE
feat: add iceberg cdc test

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: e2e_test/iceberg/main.py [hunk 1]
```diff
@@ -42,14 +42,16 @@ def init_iceberg_table(args,init_sqls):
...
+    if slt is None or slt == "":
+        return
     rw_config = args['risingwave']
     cmd = f"sqllogictest -p {rw_config['port']} -d {rw_config['db']} {slt}"
     print(f"Command line is [{cmd}]")
     subprocess.run(cmd,
                    shell=True,
                    check=True)
-    time.sleep(10)
+    time.sleep(30)
```
Comment (liurenjie1024) [inline e2e_test/iceberg/main.py:54]: Do we really need 30 seconds? How about modifying it to 15 secs?

File: integration_tests/iceberg-cdc/run_test.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+set -ex
+
+# Start test environment.
+docker-compose up -d
```
Comment (liurenjie1024) [inline integration_tests/iceberg-cdc/run_test.sh:6]:
```suggestion
docker-compose up -d --wait
```

Wait for services to be ready.

File: integration_tests/iceberg-cdc/run_test.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+# Wait for the environment to be ready.
+sleep 20;
```
Comment (liurenjie1024) [inline integration_tests/iceberg-cdc/run_test.sh:9]: Remove this

File: integration_tests/iceberg-cdc/docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    expose:
+      - 8181
+    ports:
+      - "8181:8181"
+
+  spark:
+    depends_on:
+      - minio-0
+      - rest
+    build: ./spark
```
Comment (liurenjie1024) [inline integration_tests/iceberg-cdc/docker-compose.yaml:108]: How about using images from icelake repo directly?

**Review by liurenjie1024 — COMMENTED** [review]
Generally LGTM, thanks

File: docker/docker-compose-latest.yml
[the remark this answers is not in the document]
Comment (ZENOTME) [reply docker/docker-compose-latest.yml:1]: `docker/doceker-compose.yml` use risingwave `v1.2` and we haven't support iceberg upsert in that version. How about update it to `v1.3`?

[the remark this answers is not in the document]
Comment (liurenjie1024) [reply docker/docker-compose-latest.yml:1]: I think we should upgrade it to 1.3.

File: integration_tests/iceberg-cdc/run_test.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
Comment (liurenjie1024) [inline integration_tests/iceberg-cdc/run_test.sh:1]: This is not in ci? See https://github.com/risingwavelabs/risingwave/blob/61a5bd59a31992f0c514150913e7d41c32bee246/ci/workflows/integration-tests.yml#L17

**Review by liurenjie1024 — COMMENTED** [review]
LGTM, please add it to ci.

File: integration_tests/iceberg-cdc/run_test.sh (hunk 5 above)
Comment (ZENOTME) [reply integration_tests/iceberg-cdc/run_test.sh:1]: I have add it, do we have a way to test it in this PR?

Comment (liurenjie1024) [reply integration_tests/iceberg-cdc/run_test.sh:1]: No, it's tested in daily ci. We can wait an see.

File: ci/workflows/integration-tests.yml [hunk 6]
```diff
@@ -79,6 +80,10 @@ steps:
         #    testcase: "iceberg-sink"
         #    format: "protobuf"
         #  skip: true
+        - with:
+            testcase: "iceberg-cdc"
+            format: "protobuf"
+          skip: true
```
Comment (liurenjie1024) [inline ci/workflows/integration-tests.yml:86]: What's this `skip: true` means? cc @neverchanje

Comment (ZENOTME) [reply ci/workflows/integration-tests.yml:86]: In this matrix, it will execute `iceberg-cdc,protobuf`, `iceberg-cdc,json`. We just skip one.

**Review by liurenjie1024 — APPROVED** [review]

**Pull request merged**